File: leetcode/java/leetcode/src/linkedList/_237_delete_node_in_a_linked_list.java
/**
 * https://leetcode-cn.com/problems/delete-node-in-a-linked-list/
 */
public class _237_delete_node_in_a_linked_list {

    public class ListNode {
        int val;
        ListNode next;
        ListNode(int x) {
            val = x;
        }
    }

    class Solution {
        public void deleteNode(ListNode node) {
            node.val = node.next.val;
            node.next = node.next.next;
        }
    }
}

File: leetcode/js/linkedList/_237_delete_node_in_a_linked_list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */


 function ListNode(val) {
    this.val = val;
    this.next = null;
 }


/**
 * https://leetcode-cn.com/problems/delete-node-in-a-linked-list/
 * @param {ListNode} node
 * @return {void} Do not return anything, modify node in-place instead.
 */
 var deleteNode = function(node) {
    node.val = node.next.val;
    node.next = node.next.next;
};

var deleteNode1 = function(node) {
    node.val = node.next.val, node.next = node.next.next
};

var deleteNode2 = function (node) {
    Object.assign(node, node.next)
};